### PR TITLE
Feature/oap/annotation

### PR DESCRIPTION
**Pull request by peng-yongsheng**
Use annotation to instead of the definition file.

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/endpoint/EndpointLatencyAvgIndicator.java [hunk 1]
```diff
@@ -21,12 +21,15 @@
...
+import org.apache.skywalking.oap.server.core.remote.annotation.StreamData;
 import org.apache.skywalking.oap.server.core.remote.grpc.proto.RemoteData;
-import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.core.storage.annotation.*;
 
 /**
  * @author peng-yongsheng
  */
+@StreamData
+@StorageEntity(name = "endpoint_latency_avg")
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/endpoint/EndpointLatencyAvgIndicator.java]: This class also has `#name()` method to return this name. Do you still need that method?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationListener.java [hunk 2]
```diff
@@ -16,14 +16,16 @@
...
  */
-public class IndicatorDefineLoadException extends Exception {
+public interface AnnotationListener {
+
+    Class<? extends Annotation> annotation();
 
-    public IndicatorDefineLoadException(String message, Throwable cause) {
-        super(message, cause);
-    }
+    void ownerClass(Class aClass);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationListener.java]: I feel a little strange about `ownerClass` method name. This is a method of `*Listenser`,  owner makes me think it is the owner of the listener. Clearly, it isn't. 

So I proposal, you should name this as something like `notify`, which should more make sense.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationScan.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    public void scan(Runnable callBack) throws IOException {
+        ClassPath classpath = ClassPath.from(this.getClass().getClassLoader());
+        ImmutableSet<ClassPath.ClassInfo> classes = classpath.getTopLevelClassesRecursive("org.apache.skywalking");
+        for (ClassPath.ClassInfo classInfo : classes) {
+            Class<?> aClass = classInfo.load();
+
+            for (AnnotationListener listener : listeners) {
+                if (aClass.isAnnotationPresent(listener.annotation())) {
+                    listener.ownerClass(aClass);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationScan.java]: Read this line, which will look like `listener's owner is aClass`.

**Comment by wu-sheng** [comment]
BTW, CI fails.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationListener.java (hunk 2 above)
Comment (peng-yongsheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationListener.java]: notify is a key word.

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/annotation/AnnotationListener.java]: No, it isn't.

**Pull request merged**